From 32fe5a8b9b8e31e396e52a8121fb1286f838e39a Mon Sep 17 00:00:00 2001
From: tanyulong <tanyulong@kylinos.cn>
Date: Thu, 29 Jul 2021 19:46:40 +0800
Subject: [PATCH] fix compile extern C error

---
 plugins/account/userinfo/run-passwd.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/plugins/account/userinfo/run-passwd.cpp b/plugins/account/userinfo/run-passwd.cpp
index 6407929..80da0d0 100755
--- a/plugins/account/userinfo/run-passwd.cpp
+++ b/plugins/account/userinfo/run-passwd.cpp
@@ -7,7 +7,6 @@
 #undef signals
 #endif
 
-extern "C" {
 #include <glib.h>
 #include <gio/gio.h>
 
@@ -16,7 +15,6 @@ extern "C" {
 #include <errno.h>
 #include <sys/wait.h>
 
-}
 
 #include "run-passwd.h"
 
-- 
2.27.0

