From a59a56767afb447307264827df8ae9a03b60f250 Mon Sep 17 00:00:00 2001
From: pei-jiankang <peijiankang@kylinos.cn>
Date: Wed, 9 Mar 2022 09:42:52 +0800
Subject: [PATCH] disable the str "'" of password check

---
 plugins/account/userinfo/createuserdialog.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/plugins/account/userinfo/createuserdialog.cpp b/plugins/account/userinfo/createuserdialog.cpp
index 2af9290..e35732c 100644
--- a/plugins/account/userinfo/createuserdialog.cpp
+++ b/plugins/account/userinfo/createuserdialog.cpp
@@ -318,8 +318,6 @@ void CreateUserDialog::pwdLegalityCheck(QString pwd){
 }
 
 bool CreateUserDialog::checkCharLegitimacy(QString password){
-    if (password.contains("'"))
-        return false;
     //密码不能包含非标准字符
     foreach (QChar ch, password){
         if (int(ch.toLatin1() <= 0 || int(ch.toLatin1()) > 127)){
-- 
2.33.0

