From 0f33449dc1c22e97331265a6410e9a0a5466d39c Mon Sep 17 00:00:00 2001
From: pei-jiankang <peijiankang@kylinos.cn>
Date: Wed, 6 Apr 2022 17:29:15 +0800
Subject: [PATCH] modify icon theme not display

---
 plugins/personalized/theme/theme.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/plugins/personalized/theme/theme.cpp b/plugins/personalized/theme/theme.cpp
index 893db27..59f66af 100644
--- a/plugins/personalized/theme/theme.cpp
+++ b/plugins/personalized/theme/theme.cpp
@@ -443,8 +443,9 @@ void Theme::initIconTheme() {
 
     foreach (QString themedir, themesDir.entryList(QDir::Dirs)) {
 
-        if ((Utils::isCommunity() && (!themedir.compare("ukui") || !themedir.compare("ukui-classical")))
-                || (!Utils::isCommunity() && themedir.startsWith("ukui-icon-theme-"))) {
+        if ((Utils::isCommunity() && (!themedir.compare("ukui") || !themedir.compare("ukui-classical")) )
+                || (!Utils::isCommunity() && themedir.startsWith("ukui-icon-theme-"))
+                || (!Utils::isCommunity() && (!themedir.compare("ukui") || !themedir.compare("ukui-classical") || !themedir.compare("ukui-fashion")))) {
             QDir appsDir = QDir(ICONTHEMEPATH + themedir + "/48x48/apps/");
             QDir placesDir = QDir(ICONTHEMEPATH + themedir + "/48x48/places/");
             QDir devicesDir = QDir(ICONTHEMEPATH + themedir + "/48x48/devices/");
-- 
2.33.0

