From d5fb1fd93c7733ca0a2111c349149a8fe44926d2 Mon Sep 17 00:00:00 2001
From: tanyulong <tanyulong@kylinos.cn>
Date: Mon, 12 Jul 2021 09:26:23 +0800
Subject: [PATCH] fix layout optimization

---
 plugins/system/autoboot/addautoboot.ui | 20 ++------------------
 1 file changed, 2 insertions(+), 18 deletions(-)

diff --git a/plugins/system/autoboot/addautoboot.ui b/plugins/system/autoboot/addautoboot.ui
index 4e04bfa..e778c84 100755
--- a/plugins/system/autoboot/addautoboot.ui
+++ b/plugins/system/autoboot/addautoboot.ui
@@ -255,8 +255,8 @@
              <widget class="QLineEdit" name="execLineEdit">
               <property name="minimumSize">
                <size>
-                <width>182</width>
-                <height>28</height>
+		<width>190</width>
+		<height>28</height>
                </size>
               </property>
               <property name="maximumSize">
@@ -346,22 +346,6 @@
             </property>
            </widget>
           </item>
-          <item>
-           <spacer name="verticalSpacer">
-            <property name="orientation">
-             <enum>Qt::Vertical</enum>
-            </property>
-            <property name="sizeType">
-             <enum>QSizePolicy::Fixed</enum>
-            </property>
-            <property name="sizeHint" stdset="0">
-             <size>
-              <width>20</width>
-              <height>48</height>
-             </size>
-            </property>
-           </spacer>
-          </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
-- 
2.23.0

