From 648547177ddceb27394135bb276deb18666019cf Mon Sep 17 00:00:00 2001
From: tanyulong2021 <tanyulong@kylinos.cn>
Date: Mon, 27 Feb 2023 14:37:21 +0800
Subject: [PATCH] Fix about copyright display error

---
 plugins/system/about/about.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/system/about/about.cpp b/plugins/system/about/about.cpp
index e63591c..824b781 100644
--- a/plugins/system/about/about.cpp
+++ b/plugins/system/about/about.cpp
@@ -465,7 +465,7 @@ void About::initUI(QWidget *widget)
     mActivationLayout_1->addLayout(mActivationLayout);
     mActivationLayout_1->addWidget(mActivationBtn);
 
-    QString currentyear("2022");
+    QString currentyear("2023");
     mTipLabel = new FixLabel(QString(tr("Copyright © 2009-%1 KylinSoft. All rights reserved.")).arg(currentyear) , Aboutwidget);
     mTipLabel->setContentsMargins(16 , 0 , 0 , 0);
 
-- 
2.33.0

