From a16a9dfc376d0bd23b25627084fccb2c87b9fe58 Mon Sep 17 00:00:00 2001
From: tanyulong2021 <tanyulong@kylinos.cn>
Date: Thu, 28 Jul 2022 16:51:41 +0800
Subject: [PATCH] add dependency ddcutil

---
 debian/control | 1 +
 1 file changed, 1 insertion(+)

diff --git a/debian/control b/debian/control
index c9b3f2e..3d1a8bc 100644
--- a/debian/control
+++ b/debian/control
@@ -56,6 +56,7 @@ Depends: ${shlibs:Depends},
          qml-module-qtquick-controls,
          qt5-image-formats-plugins,
          dconf-cli,
+	 ddcutil,
          kylin-software-properties(>=0.0.1-55),
 	 kylin-update-manager(>=5.4.6kord~rc5),
          language-selector-gnome,
-- 
2.33.0

