diff -Naur ukui-control-center-3.0.1/plugins/system/display/display.cpp ukui-control-center-3.0.1~/plugins/system/display/display.cpp
--- ukui-control-center-3.0.1/plugins/system/display/display.cpp	2021-09-01 16:38:36.839491260 +0800
+++ ukui-control-center-3.0.1~/plugins/system/display/display.cpp	2021-09-01 16:42:42.542192296 +0800
@@ -29,6 +29,10 @@
 #include <QDBusConnection>
 
 DisplaySet::DisplaySet(){
+    mFirstLoad = true;
+    //make sure backend start
+    requestBackend();
+    
     pluginWidget = new Widget;
     pluginName = tr("Display");
     QObject::connect(new KScreen::GetConfigOperation(), &KScreen::GetConfigOperation::finished,
@@ -87,8 +91,9 @@
         QString output = process.readAll();
         output = output.simplified();
 
-        QString command = "/usr/lib/" + output + "-linux-gnu" +"/libexec/kf5/kscreen_backend_launcher";
-        QProcess::startDetached(command);
+        //QString command = "/usr/lib/" + output + "-linux-gnu" +"/libexec/kf5/kscreen_backend_launcher";
+        QString command = "/usr/libexec/kf5/kscreen_backend_launcher";
+	QProcess::startDetached(command);
     }
 }
 
