diff -Naur ukui-control-center-3.0.1/plugins/account/userinfo/changevaliddialog.cpp ukui-control-center-3.0.1~/plugins/account/userinfo/changevaliddialog.cpp
--- ukui-control-center-3.0.1/plugins/account/userinfo/changevaliddialog.cpp	2019-05-27 15:48:18.000000000 +0800
+++ ukui-control-center-3.0.1~/plugins/account/userinfo/changevaliddialog.cpp	2021-09-03 18:48:22.196993408 +0800
@@ -155,8 +155,8 @@
     if (valid.startsWith(_name)){
         QStringList validList = valid.split(" ");
         QString lastChangeStr = validList.at(2);
-        QStringList lastChangeList = lastChangeStr.split("/");
-        lastChangeDate = QDate(QString(lastChangeList.at(2)).toInt(), QString(lastChangeList.at(0)).toInt(), QString(lastChangeList.at(1)).toInt());
+        QStringList lastChangeList = lastChangeStr.split("-");
+        lastChangeDate = QDate(QString(lastChangeList.at(0)).toInt(), QString(lastChangeList.at(1)).toInt(), QString(lastChangeList.at(2)).toInt());
         delayDays = QString(validList.at(4)).toInt();
     } else {
         delayDays = -1;
@@ -201,7 +201,7 @@
     ui->yearCombox->clear();
 
     QDate begin = QDate::currentDate().addDays(1);
-    QDate canSelect = lastChangeDate.addYears(26);
+    QDate canSelect = lastChangeDate.addYears(70);
 
     ui->yearCombox->addItem(QObject::tr("Never"), 0);
     for (int year = begin.year(); year <= canSelect.year(); year++){
